src/main.js

```diff
@@ -128,6 +128,16 @@ const trayImages = function() {
 }();
 var willAppQuit = false;
 
+// If there is already an instance, activate the window in the existing instace and quit this one
+if (app.makeSingleInstance((commandLine, workingDirectory) => {
+    // Someone tried to run a second instance, we should focus our window.
+    if (mainWindow) {
+      mainWindow.show()
```

Comment:
This might change the window state of the already running instance. 
- Try to maximize the window, then send it to the task bar (click the _ of the window), then try to start mattermost a second time. 
- The other case you need to check is maximize the window, then click the X to send it to the tray, then try to start mattermost a second time.

Something like this should work.
```if (mainWindow.isMinimized()) mainWindow.restore();
else mainWindow.show();
```